src/phpDebug.ts

```diff
@@ -438,6 +442,7 @@ class PhpDebugSession extends vscode.DebugSession {
         try {
             const fileUri = convertClientPathToDebugger(args.source.path!, this._args.pathMappings)
             const connections = Array.from(this._connections.values())
+            const hasWaitingConnections = this._waitingConnections.size > 0
```

Comment:
It is not clear to me why the existence of a waiting connection is of significance for this behavior. Could you explain?

Reply:
Well as you said, the way I'm doing it now, just skipping the connection altogether if it is blocked, is rather brittle.  So my thought was, the deadlock specifically happens because the configuration of a new connection is waiting on the configuration to be initialized, which involves calling this code to set up breakpoints.  So if `_waitingConnections` has something in it, that is when we should "care" that there could potentially be a command blocking it from continuing.

That said, if I change it's behavior to instead do a delayed update instead of just skipping the connection, this would no longer be needed.  So if I get that working I'll remove this condition as well.